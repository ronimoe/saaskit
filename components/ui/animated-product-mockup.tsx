'use client'

import { useEffect, useState } from 'react'
import { cn } from '@/lib/utils'
import { 
  Monitor, 
  Smartphone, 
  Tablet, 
  CheckCircle, 
  CreditCard, 
  Users, 
  BarChart3,
  Shield,
  Zap,
  Globe
} from 'lucide-react'

interface MockupState {
  id: string
  title: string
  description: string
  icon: React.ComponentType<{ className?: string }>
  screens: {
    desktop: string
    mobile: string
    tablet?: string
  }
  color: string
}

const mockupStates: MockupState[] = [
  {
    id: 'dashboard',
    title: 'Analytics Dashboard',
    description: 'Real-time insights and beautiful visualizations',
    icon: BarChart3,
    screens: {
      desktop: 'dashboard-desktop',
      mobile: 'dashboard-mobile'
    },
    color: 'from-blue-500 to-purple-600'
  },
  {
    id: 'auth',
    title: 'Secure Authentication',
    description: 'Multi-factor auth with social login options',
    icon: Shield,
    screens: {
      desktop: 'auth-desktop',
      mobile: 'auth-mobile'
    },
    color: 'from-green-500 to-teal-600'
  },
  {
    id: 'payments',
    title: 'Payment Processing',
    description: 'Stripe integration with subscription management',
    icon: CreditCard,
    screens: {
      desktop: 'payments-desktop',
      mobile: 'payments-mobile'
    },
    color: 'from-purple-500 to-pink-600'
  },
  {
    id: 'collaboration',
    title: 'Team Collaboration',
    description: 'Built-in tools for team management',
    icon: Users,
    screens: {
      desktop: 'team-desktop',
      mobile: 'team-mobile'
    },
    color: 'from-orange-500 to-red-600'
  }
]

interface AnimatedProductMockupProps {
  className?: string
  autoPlay?: boolean
  interval?: number
}

const MockupScreen = ({ 
  state, 
  device, 
  isActive 
}: { 
  state: MockupState
  device: 'desktop' | 'mobile' | 'tablet'
  isActive: boolean 
}) => {
  const DeviceIcon = device === 'desktop' ? Monitor : device === 'mobile' ? Smartphone : Tablet
  const StateIcon = state.icon

  const renderContent = () => {
    switch (state.id) {
      case 'dashboard':
        return (
          <div className="space-y-3">
            <div className="flex items-center justify-between">
              <div className="text-xs font-medium text-white">Analytics Overview</div>
              <div className="text-xs text-green-400">●︎ Live</div>
            </div>
            <div className="grid grid-cols-4 gap-2">
              <div className="h-8 bg-gradient-to-t from-blue-500 to-cyan-400 rounded opacity-90 animate-pulse"></div>
              <div className="h-10 bg-gradient-to-t from-purple-500 to-pink-400 rounded opacity-90 animate-pulse" style={{ animationDelay: '0.2s' }}></div>
              <div className="h-6 bg-gradient-to-t from-green-500 to-emerald-400 rounded opacity-90 animate-pulse" style={{ animationDelay: '0.4s' }}></div>
              <div className="h-9 bg-gradient-to-t from-orange-500 to-yellow-400 rounded opacity-90 animate-pulse" style={{ animationDelay: '0.6s' }}></div>
            </div>
            <div className="space-y-1">
              <div className="flex justify-between text-xs">
                <span className="text-slate-300">Revenue</span>
                <span className="text-green-400">+12.5%</span>
              </div>
              <div className="h-1 bg-slate-700 rounded-full overflow-hidden">
                <div className="h-full bg-green-400 rounded-full animate-pulse" style={{ width: '75%' }}></div>
              </div>
            </div>
          </div>
        )
      
      case 'auth':
        return (
          <div className="space-y-3">
            <div className="flex items-center gap-2">
              <Shield className="w-4 h-4 text-green-400" />
              <span className="text-xs font-medium text-white">Secure Login</span>
            </div>
            <div className="space-y-2">
              <div className="h-6 bg-slate-700 rounded border border-slate-600"></div>
              <div className="h-6 bg-slate-700 rounded border border-slate-600"></div>
              <div className="h-6 bg-blue-500 rounded text-white text-xs flex items-center justify-center">
                Sign In Securely
              </div>
            </div>
            <div className="flex items-center gap-2 text-xs text-green-400">
              <CheckCircle className="w-3 h-3" />
              <span>2FA Enabled</span>
            </div>
          </div>
        )
      
      case 'payments':
        return (
          <div className="space-y-3">
            <div className="flex items-center gap-2">
              <CreditCard className="w-4 h-4 text-purple-400" />
              <span className="text-xs font-medium text-white">Payment Portal</span>
            </div>
            <div className="space-y-2">
              <div className="flex justify-between text-xs">
                <span className="text-slate-300">Pro Plan</span>
                <span className="text-white font-medium">$29/mo</span>
              </div>
              <div className="h-2 bg-slate-700 rounded-full overflow-hidden">
                <div className="h-full bg-purple-500 rounded-full animate-pulse" style={{ width: '60%' }}></div>
              </div>
              <div className="text-xs text-purple-400">Payment processing...</div>
            </div>
          </div>
        )
      
      case 'collaboration':
        return (
          <div className="space-y-3">
            <div className="flex items-center gap-2">
              <Users className="w-4 h-4 text-orange-400" />
              <span className="text-xs font-medium text-white">Team Dashboard</span>
            </div>
            <div className="space-y-2">
              <div className="flex items-center gap-2">
                <div className="w-4 h-4 bg-orange-500 rounded-full"></div>
                <div className="text-xs text-slate-300">Alice Johnson</div>
                <div className="text-xs text-green-400">Online</div>
              </div>
              <div className="flex items-center gap-2">
                <div className="w-4 h-4 bg-blue-500 rounded-full"></div>
                <div className="text-xs text-slate-300">Bob Smith</div>
                <div className="text-xs text-yellow-400">Away</div>
              </div>
              <div className="text-xs text-orange-400">3 team members active</div>
            </div>
          </div>
        )
      
      default:
        return null
    }
  }

  if (device === 'desktop') {
    return (
      <div className={cn(
        "relative bg-slate-900 rounded-lg shadow-2xl transition-all duration-700 transform",
        isActive ? "scale-100 opacity-100 z-10" : "scale-95 opacity-60 z-0"
      )}>
        {/* Desktop Frame */}
        <div className="bg-slate-800 px-4 py-2 rounded-t-lg border-b border-slate-700">
          <div className="flex items-center gap-2">
            <div className="flex gap-1">
              <div className="w-3 h-3 bg-red-500 rounded-full"></div>
              <div className="w-3 h-3 bg-yellow-500 rounded-full"></div>
              <div className="w-3 h-3 bg-green-500 rounded-full"></div>
            </div>
            <div className="text-xs text-slate-400 ml-2">saaskit-demo.com</div>
          </div>
        </div>
        
        {/* Desktop Content */}
        <div className="p-6 h-32">
          {renderContent()}
        </div>
      </div>
    )
  }

  if (device === 'mobile') {
    return (
      <div className={cn(
        "relative bg-slate-900 rounded-2xl shadow-xl transition-all duration-700 transform border-4 border-slate-700",
        isActive ? "scale-100 opacity-100" : "scale-90 opacity-60"
      )}>
        {/* Mobile Frame */}
        <div className="bg-slate-800 h-6 rounded-t-2xl flex items-center justify-center">
          <div className="w-12 h-1 bg-slate-600 rounded-full"></div>
        </div>
        
        {/* Mobile Content */}
        <div className="p-3 h-24">
          {renderContent()}
        </div>
        
        {/* Mobile Home Indicator */}
        <div className="h-4 flex items-center justify-center">
          <div className="w-8 h-1 bg-slate-600 rounded-full"></div>
        </div>
      </div>
    )
  }

  return null
}

export function AnimatedProductMockup({
  className,
  autoPlay = true,
  interval = 4000
}: AnimatedProductMockupProps) {
  const [currentState, setCurrentState] = useState(0)
  const [isPlaying, setIsPlaying] = useState(autoPlay)

  useEffect(() => {
    if (!isPlaying) return

    const timer = setInterval(() => {
      setCurrentState((prev) => (prev + 1) % mockupStates.length)
    }, interval)

    return () => clearInterval(timer)
  }, [isPlaying, interval])

  const currentMockup = mockupStates[currentState] || mockupStates[0]

  if (!currentMockup) {
    return null
  }

  return (
    <div className={cn("relative", className)}>
      {/* Feature Indicator */}
      <div className="text-center mb-8">
        <div className="inline-flex items-center gap-3 px-4 py-2 bg-slate-900/50 backdrop-blur-sm rounded-full border border-slate-700">
          <currentMockup.icon className="w-5 h-5 text-white" />
          <div>
            <div className="text-sm font-medium text-white">{currentMockup.title}</div>
            <div className="text-xs text-slate-400">{currentMockup.description}</div>
          </div>
        </div>
      </div>

      {/* Device Mockups */}
      <div className="relative flex items-start justify-center gap-8">
        {/* Desktop Mockup */}
        <div className="relative">
          <MockupScreen 
            state={currentMockup}
            device="desktop"
            isActive={true}
          />
          
          {/* Floating Elements */}
          <div className="absolute -top-4 -right-4 w-8 h-8 bg-gradient-to-br from-blue-500 to-purple-600 rounded-full opacity-60 animate-pulse"></div>
          <div className="absolute -bottom-2 -left-2 w-6 h-6 bg-gradient-to-br from-green-500 to-teal-600 rounded-full opacity-60 animate-pulse" style={{ animationDelay: '1s' }}></div>
        </div>
        
        {/* Mobile Mockup */}
        <div className="relative">
          <MockupScreen 
            state={currentMockup}
            device="mobile"
            isActive={true}
          />
          
          {/* Connection Line */}
          <div className="absolute top-1/2 -left-8 w-8 h-0.5 bg-gradient-to-r from-slate-600 to-transparent"></div>
        </div>
      </div>

      {/* Progress Indicators */}
      <div className="flex items-center justify-center gap-2 mt-8">
        {mockupStates.map((_, index) => (
          <button
            key={index}
            onClick={() => {
              setCurrentState(index)
              setIsPlaying(false)
              setTimeout(() => setIsPlaying(autoPlay), 1000)
            }}
            className={cn(
              "w-2 h-2 rounded-full transition-all duration-300",
              currentState === index 
                ? "bg-blue-500 w-8" 
                : "bg-slate-600 hover:bg-slate-500"
            )}
          />
        ))}
      </div>

      {/* Feature Icons Floating */}
      <div className="absolute inset-0 pointer-events-none">
        <div className="absolute top-10 left-10 opacity-20">
          <Zap className="w-6 h-6 text-yellow-400 animate-pulse" />
        </div>
        <div className="absolute top-20 right-16 opacity-20">
          <Globe className="w-6 h-6 text-blue-400 animate-pulse" style={{ animationDelay: '2s' }} />
        </div>
        <div className="absolute bottom-16 left-16 opacity-20">
          <CheckCircle className="w-6 h-6 text-green-400 animate-pulse" style={{ animationDelay: '4s' }} />
        </div>
      </div>

      {/* Background Glow */}
      <div className={cn(
        "absolute inset-0 -z-10 blur-3xl opacity-30 transition-all duration-1000",
        `bg-gradient-to-br ${currentMockup.color}`
      )}></div>
    </div>
  )
}

AnimatedProductMockup.displayName = 'AnimatedProductMockup' 