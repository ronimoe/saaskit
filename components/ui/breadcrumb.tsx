'use client';

import React from 'react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { ChevronRight, Home } from 'lucide-react';
import { cn } from '@/lib/utils';

interface BreadcrumbItem {
  label: string;
  href: string;
  icon?: React.ComponentType<{ className?: string }>;
}

interface BreadcrumbProps {
  className?: string;
  separator?: React.ReactNode;
  showHome?: boolean;
  customItems?: BreadcrumbItem[];
}

/**
 * Breadcrumb Navigation Component
 * 
 * Provides contextual navigation showing the current page hierarchy.
 * Automatically generates breadcrumbs from the current route or accepts custom items.
 * 
 * Features:
 * - Automatic route-based breadcrumb generation
 * - Custom breadcrumb items support
 * - Accessible navigation with proper ARIA labels
 * - Responsive design with mobile-friendly layout
 * - Customizable separator and home icon
 */
export function Breadcrumb({ 
  className, 
  separator, 
  showHome = true, 
  customItems 
}: BreadcrumbProps) {
  const pathname = usePathname();
  
  // Generate breadcrumb items from pathname if custom items not provided
  const generateBreadcrumbs = (): BreadcrumbItem[] => {
    if (customItems) return customItems;
    
    const segments = pathname.split('/').filter(Boolean);
    const breadcrumbs: BreadcrumbItem[] = [];
    
    // Add home/dashboard if showHome is true
    if (showHome) {
      breadcrumbs.push({
        label: 'Dashboard',
        href: '/dashboard',
        icon: Home
      });
    }
    
    // Generate breadcrumbs for each path segment
    let currentPath = '';
    segments.forEach((segment, index) => {
      currentPath += `/${segment}`;
      
      // Skip if this is the dashboard segment and we already added it
      if (segment === 'dashboard' && showHome) return;
      
      // Convert segment to readable label
      const label = segment
        .split('-')
        .map(word => word.charAt(0).toUpperCase() + word.slice(1))
        .join(' ');
      
      breadcrumbs.push({
        label,
        href: currentPath
      });
    });
    
    return breadcrumbs;
  };
  
  const breadcrumbs = generateBreadcrumbs();
  
  // Don't render if no breadcrumbs or only one item
  if (breadcrumbs.length <= 1) return null;
  
  const defaultSeparator = separator || <ChevronRight className="h-4 w-4 text-muted-foreground" />;
  
  return (
    <nav 
      aria-label="Breadcrumb navigation" 
      className={cn("flex items-center space-x-1 text-sm text-muted-foreground overflow-x-auto pb-2", className)}
    >
              <ol className="flex items-center space-x-1 whitespace-nowrap">
        {breadcrumbs.map((item, index) => {
          const isLast = index === breadcrumbs.length - 1;
          const Icon = item.icon;
          
          return (
            <li key={item.href} className="flex items-center space-x-1">
              {index > 0 && (
                <span className="flex items-center" aria-hidden="true">
                  {defaultSeparator}
                </span>
              )}
              
              {isLast ? (
                // Current page - not clickable
                <span 
                  className="flex items-center space-x-1 font-medium text-foreground"
                  aria-current="page"
                >
                  {Icon && <Icon className="h-4 w-4" />}
                  <span>{item.label}</span>
                </span>
              ) : (
                // Clickable breadcrumb link
                <Link
                  href={item.href}
                  className="flex items-center space-x-1 hover:text-foreground transition-colors min-h-[44px] py-2 px-1 -mx-1 rounded-md hover:bg-accent/50"
                >
                  {Icon && <Icon className="h-4 w-4" />}
                  <span>{item.label}</span>
                </Link>
              )}
            </li>
          );
        })}
      </ol>
    </nav>
  );
}

/**
 * Breadcrumb Separator Component
 * 
 * Custom separator for breadcrumb navigation
 */
export function BreadcrumbSeparator({ 
  children, 
  className 
}: { 
  children?: React.ReactNode; 
  className?: string; 
}) {
  return (
    <span className={cn("text-muted-foreground", className)} aria-hidden="true">
      {children || <ChevronRight className="h-4 w-4" />}
    </span>
  );
}

/**
 * Breadcrumb Item Component
 * 
 * Individual breadcrumb item for custom breadcrumb compositions
 */
interface BreadcrumbItemComponentProps {
  href?: string;
  isActive?: boolean;
  children: React.ReactNode;
  className?: string;
}

export function BreadcrumbItem({ 
  href, 
  isActive = false, 
  children, 
  className 
}: BreadcrumbItemComponentProps) {
  const baseClasses = "flex items-center space-x-1 transition-colors";
  
  if (isActive) {
    return (
      <span 
        className={cn(baseClasses, "font-medium text-foreground", className)}
        aria-current="page"
      >
        {children}
      </span>
    );
  }
  
  if (href) {
    return (
      <Link
        href={href}
        className={cn(baseClasses, "text-muted-foreground hover:text-foreground", className)}
      >
        {children}
      </Link>
    );
  }
  
  return (
    <span className={cn(baseClasses, "text-muted-foreground", className)}>
      {children}
    </span>
  );
} 