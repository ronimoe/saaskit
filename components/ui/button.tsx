/**
 * Button Component
 * 
 * A versatile button component built on top of Radix UI Slot with class-variance-authority
 * for type-safe variant handling. Supports multiple variants, sizes, and can render as
 * different elements using the asChild prop.
 * 
 * @example
 * ```tsx
 * // Basic button
 * <Button>Click me</Button>
 * 
 * // Button with variant and size
 * <Button variant="destructive" size="lg">Delete</Button>
 * 
 * // Button as a link
 * <Button asChild>
 *   <Link href="/dashboard">Go to Dashboard</Link>
 * </Button>
 * ```
 */

import * as React from "react"
import { Slot } from "@radix-ui/react-slot"
import { cva, type VariantProps } from "class-variance-authority"

import { cn } from "@/lib/utils"

/**
 * Button variant configuration using class-variance-authority
 * Defines all possible button styles and their corresponding CSS classes
 */
const buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive",
  {
    variants: {
      variant: {
        default:
          "bg-primary text-primary-foreground shadow-xs hover:bg-primary/90",
        destructive:
          "bg-destructive text-white shadow-xs hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60",
        outline:
          "border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50",
        secondary:
          "bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80",
        ghost:
          "hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50",
        link: "text-primary underline-offset-4 hover:underline",
      },
      size: {
        default: "h-9 px-4 py-2 has-[>svg]:px-3",
        sm: "h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5",
        lg: "h-10 rounded-md px-6 has-[>svg]:px-4",
        icon: "size-9",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  }
)

/**
 * Button component props interface
 */
interface ButtonProps extends 
  React.ComponentProps<"button">,
  VariantProps<typeof buttonVariants> {
  /**
   * When true, the button will render as a Slot component,
   * allowing it to merge with its child component
   * @default false
   */
  asChild?: boolean
}

/**
 * Button Component
 * 
 * @param className - Additional CSS classes to apply
 * @param variant - Button style variant (default, destructive, outline, secondary, ghost, link)
 * @param size - Button size (default, sm, lg, icon)
 * @param asChild - Whether to render as a Slot component
 * @param props - Additional button props
 */
function Button({
  className,
  variant,
  size,
  asChild = false,
  ...props
}: ButtonProps) {
  const Comp = asChild ? Slot : "button"

  return (
    <Comp
      data-slot="button"
      className={cn(buttonVariants({ variant, size, className }))}
      {...props}
    />
  )
}

export { Button, buttonVariants }
