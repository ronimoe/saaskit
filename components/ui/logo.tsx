import React from 'react';
import Image from 'next/image';
import { useBrandLogos } from '@/components/providers/brand-provider';
import { cn } from '@/lib/utils';

interface LogoProps {
  type?: 'primary' | 'secondary' | 'icon';
  className?: string;
  width?: number;
  height?: number;
  priority?: boolean;
  sizes?: string;
  style?: React.CSSProperties;
  onClick?: () => void;
}

export function Logo({
  type = 'primary',
  className,
  width,
  height,
  priority = false,
  sizes,
  style,
  onClick,
}: LogoProps) {
  const { getLogoProps } = useBrandLogos();
  const logoProps = getLogoProps(type);

  return (
    <Image
      src={logoProps.src}
      alt={logoProps.alt}
      width={width ?? logoProps.width ?? 120}
      height={height ?? logoProps.height ?? 40}
      priority={priority}
      sizes={sizes}
      className={cn('object-contain', className)}
      style={style}
      onClick={onClick}
    />
  );
}

// Specialized logo components for common use cases
export function BrandLogo({ className, ...props }: Omit<LogoProps, 'type'>) {
  return <Logo type="primary" className={cn('h-8 w-auto', className)} {...props} />;
}

export function BrandIcon({ className, ...props }: Omit<LogoProps, 'type'>) {
  return <Logo type="icon" className={cn('h-6 w-6', className)} {...props} />;
}

export function BrandMark({ className, ...props }: Omit<LogoProps, 'type'>) {
  return <Logo type="secondary" className={cn('h-8 w-8', className)} {...props} />;
} 