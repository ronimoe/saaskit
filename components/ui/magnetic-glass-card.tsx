'use client'

import { useEffect, useRef } from 'react'
import { GlassCard, GlassCardProps } from './glass-card'
import { cn } from '@/lib/utils'

interface MagneticGlassCardProps extends GlassCardProps {
  /** Enable magnetic hover effect */
  magnetic?: boolean
  /** Strength of the magnetic effect (0-1) */
  magneticStrength?: number
  /** Enable magnetic glow effect */
  magneticGlow?: boolean
}

export function MagneticGlassCard({ 
  magnetic = true,
  magneticStrength = 0.3,
  magneticGlow = false,
  className,
  children,
  ...props 
}: MagneticGlassCardProps) {
  const elementRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    if (!magnetic || !elementRef.current) return

    const element = elementRef.current
    
    const handleMouseMove = (event: MouseEvent) => {
      if (!element) return
      
      // Check for reduced motion preference
      if (window.matchMedia('(prefers-reduced-motion: reduce)').matches) return

      const rect = element.getBoundingClientRect()
      const x = event.clientX - rect.left - rect.width / 2
      const y = event.clientY - rect.top - rect.height / 2
      
      const moveX = (x / rect.width) * magneticStrength * 50
      const moveY = (y / rect.height) * magneticStrength * 50
      
      element.style.setProperty('--mouse-x', `${x}px`)
      element.style.setProperty('--mouse-y', `${y}px`)
      element.style.transform = `translate(${moveX}px, ${moveY}px)`
    }

    const handleMouseLeave = () => {
      if (!element) return
      element.style.transform = 'translate(0px, 0px)'
    }

    element.addEventListener('mousemove', handleMouseMove)
    element.addEventListener('mouseleave', handleMouseLeave)

    return () => {
      element.removeEventListener('mousemove', handleMouseMove)
      element.removeEventListener('mouseleave', handleMouseLeave)
    }
  }, [magnetic, magneticStrength])

  return (
    <GlassCard
      ref={elementRef}
      className={cn(
        magnetic && 'magnetic-hover',
        magneticGlow && 'magnetic-glow',
        'transition-transform duration-300 ease-out',
        className
      )}
      {...props}
    >
      {children}
    </GlassCard>
  )
}

MagneticGlassCard.displayName = 'MagneticGlassCard' 