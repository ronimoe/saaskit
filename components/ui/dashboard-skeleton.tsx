import React from 'react';
import { Card, CardContent, CardHeader } from '@/components/ui/card';
import { Skeleton } from '@/components/ui/skeleton';

/**
 * Dashboard Skeleton Components
 * 
 * Provides loading skeleton states for various dashboard content areas
 * to improve perceived performance during data fetches.
 */

export function MetricsCardSkeleton() {
  return (
    <Card>
      <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
        <Skeleton role="presentation" className="h-4 w-20" />
        <Skeleton role="presentation" className="h-4 w-4 rounded-full" />
      </CardHeader>
      <CardContent>
        <Skeleton role="presentation" className="h-8 w-16 mb-2" />
        <div className="flex items-center space-x-2">
          <Skeleton role="presentation" className="h-5 w-10 rounded-full" />
          <Skeleton role="presentation" className="h-3 w-32" />
        </div>
      </CardContent>
    </Card>
  );
}

export function QuickActionCardSkeleton() {
  return (
    <Card>
      <CardHeader>
        <div className="flex items-center space-x-2">
          <Skeleton role="presentation" className="h-5 w-5 rounded-full" />
          <Skeleton role="presentation" className="h-5 w-24" />
        </div>
        <Skeleton role="presentation" className="h-4 w-48 mt-2" />
      </CardHeader>
      <CardContent>
        <Skeleton role="presentation" className="h-10 w-full rounded-md" />
      </CardContent>
    </Card>
  );
}

export function RecentActivitySkeleton() {
  return (
    <Card>
      <CardHeader>
        <Skeleton role="presentation" className="h-6 w-32 mb-2" />
        <Skeleton role="presentation" className="h-4 w-56" />
      </CardHeader>
      <CardContent>
        <div className="space-y-4">
          {[...Array(3)].map((_, i) => (
            <div key={i} className="flex items-center space-x-4">
              <Skeleton role="presentation" className="w-2 h-2 rounded-full" />
              <div className="flex-1 space-y-2">
                <Skeleton role="presentation" className="h-4 w-48" />
                <Skeleton role="presentation" className="h-3 w-20" />
              </div>
            </div>
          ))}
        </div>
      </CardContent>
    </Card>
  );
}

export function TableSkeleton({ rows = 5, columns = 4 }: { rows?: number; columns?: number }) {
  return (
    <Card>
      <CardHeader>
        <Skeleton role="presentation" className="h-6 w-32 mb-2" />
        <Skeleton role="presentation" className="h-4 w-64" />
      </CardHeader>
      <CardContent>
        <div className="space-y-4">
          {/* Table Header */}
          <div className="grid gap-4" style={{ gridTemplateColumns: `repeat(${columns}, 1fr)` }}>
            {[...Array(columns)].map((_, i) => (
              <Skeleton role="presentation" key={i} className="h-4 w-20" />
            ))}
          </div>
          {/* Table Rows */}
          {[...Array(rows)].map((_, rowIndex) => (
            <div key={rowIndex} className="grid gap-4" style={{ gridTemplateColumns: `repeat(${columns}, 1fr)` }}>
              {[...Array(columns)].map((_, colIndex) => (
                <Skeleton role="presentation" key={colIndex} className="h-4 w-full" />
              ))}
            </div>
          ))}
        </div>
      </CardContent>
    </Card>
  );
}

export function FormSkeleton() {
  return (
    <Card>
      <CardHeader>
        <Skeleton role="presentation" className="h-6 w-40 mb-2" />
        <Skeleton role="presentation" className="h-4 w-72" />
      </CardHeader>
      <CardContent className="space-y-6">
        {/* Form Fields */}
        {[...Array(4)].map((_, i) => (
          <div key={i} className="space-y-2">
            <Skeleton role="presentation" className="h-4 w-24" />
            <Skeleton role="presentation" className="h-10 w-full rounded-md" />
          </div>
        ))}
        {/* Submit Button */}
        <div className="flex justify-end space-x-2">
          <Skeleton role="presentation" className="h-10 w-20 rounded-md" />
          <Skeleton role="presentation" className="h-10 w-24 rounded-md" />
        </div>
      </CardContent>
    </Card>
  );
}

export function ProfileSkeleton() {
  return (
    <div className="space-y-6">
      {/* Profile Header */}
      <Card>
        <CardContent className="pt-6">
          <div className="flex items-center space-x-4">
            <Skeleton role="presentation" className="h-16 w-16 rounded-full" />
            <div className="space-y-2">
              <Skeleton role="presentation" className="h-6 w-32" />
              <Skeleton role="presentation" className="h-4 w-48" />
            </div>
          </div>
        </CardContent>
      </Card>
      
      {/* Profile Form */}
      <FormSkeleton />
    </div>
  );
}

export function BillingSkeleton() {
  return (
    <div className="space-y-6">
      {/* Current Plan */}
      <Card>
        <CardHeader>
          <Skeleton role="presentation" className="h-6 w-32 mb-2" />
          <Skeleton role="presentation" className="h-4 w-56" />
        </CardHeader>
        <CardContent>
          <div className="flex items-center justify-between">
            <div className="space-y-2">
              <Skeleton role="presentation" className="h-5 w-24" />
              <Skeleton role="presentation" className="h-4 w-32" />
            </div>
            <Skeleton role="presentation" className="h-10 w-28 rounded-md" />
          </div>
        </CardContent>
      </Card>

      {/* Payment History */}
      <TableSkeleton rows={6} columns={5} />
    </div>
  );
}

export function DashboardSkeleton() {
  return (
    <div className="space-y-8">
      {/* Welcome Section Skeleton */}
      <div className="space-y-2">
        <Skeleton role="presentation" className="h-9 w-80" />
        <Skeleton role="presentation" className="h-5 w-96" />
      </div>

      {/* Metrics Grid Skeleton */}
      <div className="grid gap-4 md:grid-cols-2 lg:grid-cols-4">
        {[...Array(4)].map((_, i) => (
          <MetricsCardSkeleton key={i} />
        ))}
      </div>

      {/* Quick Actions Skeleton */}
      <div className="space-y-4">
        <Skeleton role="presentation" className="h-7 w-32" />
        <div className="grid gap-4 md:grid-cols-3">
          {[...Array(3)].map((_, i) => (
            <QuickActionCardSkeleton key={i} />
          ))}
        </div>
      </div>

      {/* Recent Activity Skeleton */}
      <RecentActivitySkeleton />
    </div>
  );
} 