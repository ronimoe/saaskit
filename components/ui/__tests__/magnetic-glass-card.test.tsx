import { render, screen, fireEvent } from '@testing-library/react';
import { MagneticGlassCard } from '@/components/ui/magnetic-glass-card';

// Mock window.matchMedia
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

describe('MagneticGlassCard', () => {
  it('should render children correctly', () => {
    render(
      <MagneticGlassCard>
        <div>Hello</div>
      </MagneticGlassCard>
    );
    expect(screen.getByText('Hello')).toBeInTheDocument();
  });

  it('should apply transform style on mouse move', () => {
    render(<MagneticGlassCard>Card</MagneticGlassCard>);
    const card = screen.getByText('Card');
    
    fireEvent.mouseMove(card, { clientX: 10, clientY: 10 });
    expect(card.style.transform).not.toBe('translate(0px, 0px)');
  });

  it('should reset transform style on mouse leave', () => {
    render(<MagneticGlassCard>Card</MagneticGlassCard>);
    const card = screen.getByText('Card');
    
    fireEvent.mouseMove(card, { clientX: 10, clientY: 10 });
    fireEvent.mouseLeave(card);
    expect(card.style.transform).toBe('translate(0px, 0px)');
  });

  it('should not apply magnetic effect when disabled', () => {
    render(<MagneticGlassCard magnetic={false}>Card</MagneticGlassCard>);
    const card = screen.getByText('Card');
    
    fireEvent.mouseMove(card, { clientX: 10, clientY: 10 });
    expect(card.style.transform).toBe('');
  });

  it('should match snapshot', () => {
    const { container } = render(<MagneticGlassCard />);
    expect(container).toMatchSnapshot();
  });
}); 