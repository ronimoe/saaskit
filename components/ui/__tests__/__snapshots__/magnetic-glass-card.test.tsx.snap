// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagneticGlassCard should match snapshot 1`] = `
<div>
  <div
    class="relative overflow-hidden backdrop-blur-md @supports not (backdrop-filter: blur(1px)) { } border bg-white/20 dark:bg-black/20 supports-[backdrop-filter]:bg-white/15 supports-[backdrop-filter]:dark:bg-black/15 border-white/30 dark:border-white/20 rounded-xl p-6 shadow-xl shadow-black/10 dark:shadow-black/30 before:absolute before:inset-0 before:rounded-[inherit] before:border before:border-white/20 before:dark:border-white/10 magnetic-hover transition-transform duration-300 ease-out"
  />
</div>
`;
